src/core_ocean/shared/mpas_ocn_vel_forcing_topographic_wave_drag.F

```diff
@@ -0,0 +1,192 @@
+   implicit none
+   private
+   save
+
+   !--------------------------------------------------------------------
+   !
+   ! Public parameters
+   !
+   !--------------------------------------------------------------------
+
+   !--------------------------------------------------------------------
+   !
+   ! Public member functions
+   !
+   !--------------------------------------------------------------------
+
+   public :: ocn_vel_forcing_topographic_wave_drag_tend, &
+             ocn_vel_forcing_topographic_wave_drag_init
+
+   !--------------------------------------------------------------------
+   !
+   ! Private module variables
+   !
+   !--------------------------------------------------------------------
+
+   logical :: topographicWaveDragOn
+   real (kind=RKIND) :: topographicWaveDragCoeff
+
+!***********************************************************************
+
+contains
+
+!***********************************************************************
+!
+!  routine ocn_vel_forcing_topographic_wave_drag_tend
+!
+!> \brief   Computes tendency term from topographic wave drag
+!> \author  Nairita Pal
+!> \date    15 Oct 2020
+!> \details
+!>  This routine computes the topographic wave drag tendency for momentum
+!>  based on current state.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_vel_forcing_topographic_wave_drag_tend(normalVelocity, & !{{{
+                     tend, err)
+
+      !-----------------------------------------------------------------
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(in) :: &
+         normalVelocity   !< Input: velocity
+
+      !-----------------------------------------------------------------
+      !
+      ! input/output variables
+      !
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(inout) :: &
+         tend          !< Input/Output: velocity tendency
+
+      !-----------------------------------------------------------------
+      !
+      ! output variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      integer :: iEdge, k
+
+      err = 0
+      if ( .not. topographicWaveDragOn ) return
+
+      call mpas_timer_start('vel topographic wave drag')
+
+      !$omp do schedule(runtime) private(k)
+      do iEdge = 1, nEdgesOwned
+        k =  maxLevelEdgeTop(iEdge)
+        ! topographic wave drag term:
+        ! du/dt = ... 1/rinv * u
+        ! rinv is the e-folding time use in HyCOM. Here
+        ! topographic_wave_drag = 1/rinv
+        if (k>0) then
```

Comment:
@philipwjones the boundary edges between water and land (i.e. no cells) have `maxLevelEdgeTop(iEdge)=0`, so this if statement was needed within the edge loop to prevent accessing k=0. Is there a more performant way to do this?

Reply:
Probably not worth worrying about for this small loop and the alternatives are probably less robust, so I think ok to keep here.